pkg/resource/deploy/source_eval.go

```diff
@@ -2147,6 +2180,35 @@ func downgradeOutputValues(v resource.PropertyMap) resource.PropertyMap {
 	return result
 }
 
+func upgradeOutputValues(
+	v resource.PropertyMap, propertyDependencies map[resource.PropertyKey][]resource.URN,
+) resource.PropertyMap {
+	// We assume that by the time this is being called we've upgraded all Secret/Computed values to outputs. We just
+	// need to add the dependency information from propertyDependencies.
+
+	result := make(resource.PropertyMap)
+	for k, pv := range v {
+		if deps, has := propertyDependencies[k]; has {
+			var output resource.Output
```

Comment:
This isn't _quite_ right, we should see if `pv` already defines the same dependencies as `deps` (because it's already an output value, or has nested output values)